client/components/suggestions/index.jsx

```diff
@@ -0,0 +1,248 @@
+			currentSuggestion: event.target.textContent,
+		} );
+	},
+
+	removeEmptySuggestions: function( suggestions ) {
+		const hasValues = x => x.length > 0;
+		return pickBy( suggestions, hasValues );
+	},
+
+	narrowDown: function( input ) {
+		const [ taxonomy, filter ] = input.split( ':' );
+		if ( taxonomy === '' ) {
+			// empty string or just ":" or ":filter" -
+			// TODO: just show welcome screen
+			return {};
+		}
+
+		let limit = 3; //default limit, changed to 5 for single taxonomy
+		let terms; //terms that we will use to create suggestions
+		let filterTerm; // part of input that will be used for filtering
+
+		if ( filter !== undefined ) {
+			// this means that we have at least taxonomy:
+			// so check if this is a correct taxonomy
+			if ( this.props.terms.hasOwnProperty( taxonomy ) ) {
+				//so we will only filter elements from this taxonomy
+				terms = pick( this.props.terms, taxonomy );
+				//limit to 5 suggestions
+				limit = 5;
+			} else {
+				// not a valid taxonomy
+				// TODO tell something to the user
+				return {};
+			}
+			filterTerm = filter;
+		} else {
+			// we just have one word so treat is as a search terms
+			filterTerm = taxonomy;
+			terms = this.props.terms;
+		}
+
+		const filtered = {};
+
+		// store filtering term for highlithing
+		this.setState( { filterTerm: filterTerm } );
+
+		for ( const key in terms ) {
+			if ( ! this.props.terms.hasOwnProperty( key ) ) {
+				continue;
+			}
+
+			filtered[ key ] = terms[ key ].filter(
+				term => term.indexOf( filterTerm ) !== -1
+			).splice( 0, limit );
+		}
+
+		return this.removeEmptySuggestions( filtered );
+	},
+
+	createTaxonomySuggestionsArray: function( suggestions ) {
+		const taxonomySuggestionsArray = [];
+
+		for ( const key in suggestions ) {
+			if ( ! suggestions.hasOwnProperty( key ) ) {
+				continue;
+			}
+			taxonomySuggestionsArray.push( ... suggestions[ key ].map( value => key + ':' + value ) );
+		}
+
+		return taxonomySuggestionsArray;
+	},
+
+	createTextWithHighlight: function( text, highlighed_text ) {
+		const re = new RegExp( '(' + highlighed_text + ')', 'g' );
+		const parts = text.split( re );
+		const token = parts.map( ( part, i ) => {
+			const key = text + i;
+			if ( part === highlighed_text ) {
+				return <span key={ key } className="suggestions__value-emphasis" >{ part }</span>;
+			}
+			return <span key={ key }className="suggestions__value-normal" >{ part }</span>;
+		} );
+
+		return token;
+	},
+
+	createSuggestions: function( suggestions ) {
+		let noOfSuggestions = 0;
+		const rendered = [];
+
+		for ( const key in suggestions ) {
+			if ( ! suggestions.hasOwnProperty( key ) ) {
+				continue;
+			}
+
+			//Add header
+			rendered.push(
+				<div key={ key }>
+					<span className="suggestions__category">{ key }</span>
+					<span className="suggestions__category-counter">
+						{ ' ' + suggestions[ key ].length + ' of ' + this.props.terms[ key ].length }
```

Comment:
This should be `i18n`ized (which also means the string has to be formatted using placeholders instead of just concatenation).